utils/gcloud.go

```diff
@@ -0,0 +1,73 @@
+package utils
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+const (
+	GcloudBucket   = "GOOGLE_CLOUD_BUCKET"
+	GcloudProject  = "GOOGLE_CLOUD_PROJECT"
+	GcloudAppCreds = "GOOGLE_APPLICATION_CREDENTIALS"
+)
+
+func isgcloudConfigOK() error {
+	// Check if project exists
+	_, err := IsEnvVarNonEmpty(GcloudProject)
+	if err != nil {
+		return err
+	}
+
+	// Check if bucket exists
+	_, err = IsEnvVarNonEmpty(GcloudBucket)
+	if err != nil {
+		return err
+	}
+
+	// Check if creds file exists
+	credsFilePath, err := IsEnvVarNonEmpty(GcloudAppCreds)
+	if err != nil {
+		return err
+	}
+	stat, err := os.Stat(credsFilePath)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("path %q specified by GOOGLE_APPLICATION_CREDENTIALS environment variable does not exist", credsFilePath)
+	}
+	if stat.IsDir() {
+		return fmt.Errorf("path %q specified by GOOGLE_APPLICATION_CREDENTIALS environment variable is a dictory", credsFilePath)
+	}
+	return nil
+}
+
+func WriteToBucket(object, fileName string) error {
+	err := isgcloudConfigOK()
```

Comment:
The various google service clients already do most of this checking. Just try to use the client and let it fail.

Reply:
I think it's better to proactively check for all the configs before app starts running, so moving it to main instead.